### PR TITLE
Blog App - Auth

### PR DESCRIPTION
**Pull request by raminka13**
- Install CanCanCan in your project.
- Add a `role` column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User', counter_cache: :posts_counter
-  has_many :comments, counter_cache: :comments_counter
-  has_many :likes, counter_cache: :likes_counter
+  has_many :comments, dependent: :delete_all, counter_cache: :comments_counter
+  has_many :likes, dependent: :delete_all, counter_cache: :likes_counter
 
   validates :title, presence: true, length: { maximum: 250 }
   validates :text, presence: true, length: { maximum: 500 }
```
Comment (GzimAsani) [inline app/models/post.rb:7]: - [ ] Kindly add  `load_and_authorize_resource`  so CanCanCan is available to your controller

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello  @raminka13  and @jaferIdrees                 :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job adding CanCanCan on your gemfile :+1:  :+1: 

 ## Required Changes 

- [ ] In future cases kindly remember to ask for a review request from the same MR [oldMr](https://github.com/jaferIdrees/blog_app/pull/9)
- [ ] Check the comments below :arrow_double_down: 







 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (raminka13) [reply app/models/post.rb:7]: @GzimAsani  That line of code is required in the controller already and it's working as intended. Of course if you run it in the rails server you can see that. You are not supposed to add logic in the model. All requirements are met.

Comment (GzimAsani) [reply app/models/post.rb:7]: @raminka13, I sincerely apologize for my honest mistake. I confused this PR with the old one when I was checking the previous review. All of your requirements are met!

**Pull request merged**